### PR TITLE
Document Routes Manifest

### PR DESCRIPTION
**Pull request by Skn0tt**
CC https://github.com/blitz-js/blitz/pull/2042

### CODE REVIEW

**Comment by Skn0tt** [comment]
A quick draft of what the docs on https://github.com/blitz-js/blitz/pull/2042 could look like. Do we position this as "just another way of doing things" or "the new normal" for Blitz projects?

**Comment by JuanM04** [comment]
@Skn0tt maybe you should open a discussion

**Comment by flybayer** [comment]
@Skn0tt @JuanM04 go with "the new normal". People are really excited about this feature

File: app/pages/docs/route-manifest.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+title: Route Manifest
+sidebar_label: Route Manifest
+---
```
Comment (flybayer) [inline app/pages/docs/route-manifest.mdx:4]: Can you add few more details on this page?

- Exactly when Routes is auto generated
- How to manually generate it

File: app/pages/docs/link.mdx [hunk 2]
```diff
@@ -5,21 +5,21 @@ sidebar_label: <Link>
...
     <ul>
       <li>
-        <Link href="/">
+        <Link href={Routes.Home}>
           <a>Home</a>
         </Link>
       </li>
       <li>
-        <Link href="/about">
+        <Link href={Routes.About}>
```
Comment (flybayer) [inline app/pages/docs/link.mdx:22]: Let's update this section to show both the regular string approach and the Routes approach

File: app/pages/docs/link.mdx [hunk 3]
```diff
@@ -33,7 +33,7 @@ export default Home
 `Link` accepts the following props:
 
 - `href` - The path inside `pages` directory. This is the only required
-  prop
+  prop. Examples show usage of [Route Manifest](./route-manifest)
```
Comment (flybayer) [inline app/pages/docs/link.mdx:36]: Let's pull this snippet out into it's own line above the "Link accepts the following props" line.

And say something like 
``` 
The `Routes` object imported from `blitz` is automatically generated based on the pages in your app. If your page component is named `About`, then you can link to it with `Routes.About`. For more information, see the [Route Manifest documentation](./route-manifest)
```

File: app/pages/docs/link.mdx [hunk 4]
```diff
@@ -73,7 +73,7 @@ Here's an example of how to create a list of links:
 const pids = ["id1", "id2", "id3"]
 {
   pids.map((pid) => (
-    <Link href={`/post/${pid}`}>
+    <Link href={Routes.Post({pid})}>
```
Comment (flybayer) [inline app/pages/docs/link.mdx:76]: Let's update this section to show both the string way and the `Routes` way.

File: app/pages/docs/link.mdx [hunk 5]
```diff
@@ -5,21 +5,23 @@ sidebar_label: <Link>
...
-import {Link} from "blitz"
+import {Link, Routes} from "blitz"
 
 function Home() {
   return (
     <ul>
       <li>
-        <Link href="/">
+        <Link href={Routes.Home()}> // using Route manifest
+        <Link href="/"> // using location
```
Comment (flybayer) [inline app/pages/docs/link.mdx:18]:
```suggestion
        <Link href={Routes.Home()}> // Option 1: use Route manifest
        <Link href="/"> // Option 2: use string or location object
```

File: app/pages/docs/link.mdx [hunk 6]
```diff
@@ -5,21 +5,23 @@ sidebar_label: <Link>
...
           <a>Home</a>
         </Link>
       </li>
       <li>
-        <Link href="/about">
+        <Link href={Routes.About()}> // using Route manifest
+        <Link href="/about"> // using location
```
Comment (flybayer) [inline app/pages/docs/link.mdx:24]:
```suggestion
        <Link href={Routes.About()}> // Option 1: use Route manifest
        <Link href="/about"> // Option 2: use string or location object
```

**Review by flybayer — DISMISSED** [review]

**Pull request merged**